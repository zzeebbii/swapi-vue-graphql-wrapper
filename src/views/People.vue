<template>
  <div>
    <Search @onSearch="onSearch" />
    <div class="cards">
      <Card
        v-for="person in people"
        :title="person.name"
        :info="`Height: ${person.height} | Mass: ${person.mass}`"
        :key="person.name"
      />
    </div>
    <div v-if="loading" class="loader"></div>
  </div>
</template>

<script>
// @ is an alias to /src
import Card from "@/components/Card.vue";
import Search from "@/components/Search.vue";
import {
  BASE_URL,
  ALL_PEOPLE_QUERY,
  SEARCH_PERSON_QUERY
} from "@/utils/constants";
import axios from "axios";

export default {
  name: "People",
  apollo: {
    people: {
      query: ALL_PEOPLE_QUERY
    },
    person: {
      query: SEARCH_PERSON_QUERY,
      variables() {
        return {
          name: this.personToSearch
        };
      },
      skip() {
        return this.skipQuery;
      },
      manual: true,
      result({ data, loading }) {
        if (!loading) {
          this.people = data.personByName;
        }
      }
    }
  },
  components: {
    Card,
    Search
  },
  data: function() {
    return {
      people: [],
      loading: 0,
      personToSearch: "",
      skipQuery: true
    };
  },
  methods: {
    onSearch(name) {
      this.personToSearch = name;
      this.skipQuery = false;
      this.$apollo.queries.person.refetch();
    }
  }
};
</script>
