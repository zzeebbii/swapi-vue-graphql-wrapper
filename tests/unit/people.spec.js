import { shallowMount } from "@vue/test-utils";
import People from "@/views/People.vue";

describe("People.vue", () => {
  it("should render with data", function() {
    const wrapper = shallowMount(People, {
      data() {
        return {
          people: [
            {
              name: "test",
              height: 192,
              mass: 1000
            }
          ]
        };
      }
    });
    expect(wrapper).toMatchSnapshot();
  });
});
