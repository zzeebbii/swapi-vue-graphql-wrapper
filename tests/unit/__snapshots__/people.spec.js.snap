// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`People.vue should render with data 1`] = `
<div>
  <search-stub url="http://google.com"></search-stub>
  <div class="all-people">
    <card-stub title="test" info="Height: 192 | Mass: 1000"></card-stub>
  </div>
</div>
`;
